src/main/java/org/elasticsearch/gateway/TransportNodesListGatewayStartedShards.java

```diff
@@ -59,8 +62,8 @@ public TransportNodesListGatewayStartedShards(Settings settings, ClusterName clu
         this.nodeEnv = env;
     }
 
-    public ActionFuture<NodesGatewayStartedShards> list(ShardId shardId, String[] nodesIds, @Nullable TimeValue timeout) {
-        return execute(new Request(shardId, nodesIds).timeout(timeout));
+    public ActionFuture<NodesGatewayStartedShards> list(ShardId shardId, String indexUUID, String[] nodesIds, @Nullable TimeValue timeout) {
+        return execute(new Request(shardId, indexUUID, Arrays.asList(nodesIds)).timeout(timeout));
```

Comment:
I think the nodes can stay as an array. The constructor just converts to an array again.